NEFI_functions/check_neon_products.r

```diff
@@ -0,0 +1,55 @@
+#This function takes a data frame that has site and year_month columns and a vector of NEON data products of interest.
+#It returns a dataframe of unique site X year_month combinations from your data and whether or not there are data for it.
+
+#testing function.
+#clear R environment, load packages.
+#rm(list=ls())
+#library(nneo)
+#library(data.table)
+#map <- readRDS('/fs/data3/caverill/NEFI_microbial/map_otu/16S_map_clean.rds')
+#of_interest <- c('DP1.10078.001','DP1.10086.001')
+#test <- check_neon_products(map,of_interest)
+#head(test)
+
+check_neon_products <- function(some_data,neon_products){
+    #isolate site data on interest
+    some_data <- data.table(some_data)
+    sites <- some_data[,.(site,year_month)]
+    sites$site_year_month <- paste(sites$site,sites$year_month,sep='_')
+    setkey(sites,site_year_month)
+    sites <- unique(sites)
+    
+    #NEON product loop
+    product_list <- list()
+    for(j in 1:length(neon_products)){
```

Comment:
general recommendation in loops to use seq_along and seq_len instead of 1:n as the latter behaves in unintuitive ways if n < 1 (e.g. if length == 0), while the former will just skip the loop